vllm/core/scheduler.py

```diff
@@ -350,14 +404,32 @@ def _schedule_decodes(
             else:
                 logger.debug(f"append slot for {seq_group}")
                 self._append_slots(seq_group, blocks_to_copy)
-                seq_groups.append(
-                    ScheduledSequenceGroup(seq_group=seq_group,
-                                           token_chunk_size=1))
+                is_prefill = seq_group.is_prefill()
+                # NOTE(sang): It assumes the model execution never fails.
+                # This is the general assumption in vllm.
+                seq_group.update_num_computed_tokens(num_running_tokens)
+                if is_prefill:
+                    prefill_seq_groups.append(
+                        ScheduledSequenceGroup(
+                            seq_group=seq_group,
+                            token_chunk_size=num_running_tokens))
+                else:
+                    assert num_running_tokens == 1, num_running_tokens
```

Comment:
would this break spec decode because the rejection sampling takes multiple token @cadedaniel

Reply:
I think when the feature is working e2e, we can remove the assert?